mmv1/third_party/terraform/resources/resource_dns_record_set.go

```diff
@@ -0,0 +1,389 @@
+							return ipv6AddressDiffSuppress(k, old, new, d)
+						}
+						return false
+					},
+				},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return strings.ToLower(strings.Trim(old, `"`)) == strings.ToLower(strings.Trim(new, `"`))
+				},
+				Description: `The string data for the records in this record set whose meaning depends on the DNS type. For TXT record, if the string data contains spaces, add surrounding \" if you don't want your string to get split on spaces. To specify a single record value longer than 255 characters such as a TXT record for DKIM, add \"\" inside the Terraform configuration string (e.g. "first255characters\"\"morecharacters").`,
+			},
+
+			"ttl": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: `The time-to-live of this record set (seconds).`,
+			},
+
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The DNS record set type.`,
+			},
+
+			"project": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The ID of the project in which the resource belongs. If it is not provided, the provider project is used.`,
+			},
+		},
+		UseJSONNumber: true,
+	}
+}
+
+func resourceDnsRecordSetCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	userAgent, err := generateUserAgentString(d, config.userAgent)
+	if err != nil {
+		return err
+	}
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+	zone := d.Get("managed_zone").(string)
+	rType := d.Get("type").(string)
+
+	// Build the change
+	chg := &dns.Change{
+		Additions: []*dns.ResourceRecordSet{
+			{
+				Name:    name,
+				Type:    rType,
+				Ttl:     int64(d.Get("ttl").(int)),
+				Rrdatas: rrdata(d),
+			},
+		},
+	}
+
+	// The terraform provider is authoritative, so what we do here is check if
+	// any records that we are trying to create already exist and make sure we
+	// delete them, before adding in the changes requested.  Normally this would
+	// result in an AlreadyExistsError.
+	log.Printf("[DEBUG] DNS record list request for %q", zone)
+	res, err := config.NewDnsClient(userAgent).ResourceRecordSets.List(project, zone).Do()
+	if err != nil {
+		return fmt.Errorf("Error retrieving record sets for %q: %s", zone, err)
+	}
+	var deletions []*dns.ResourceRecordSet
+
+	for _, record := range res.Rrsets {
+		if record.Type != rType || record.Name != name {
+			continue
+		}
+		deletions = append(deletions, record)
+	}
+	if len(deletions) > 0 {
+		chg.Deletions = deletions
+	}
+
+	log.Printf("[DEBUG] DNS Record create request: %#v", chg)
+	chg, err = config.NewDnsClient(userAgent).Changes.Create(project, zone, chg).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating DNS RecordSet: %s", err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", zone, name, rType))
```

Comment:
Thinking out loud: The `id` change is safe because we don't use it in `Read` and neither did the old implementation (so users are safely able to downgrade if needed)